emu/src/ppu.rs

```diff
@@ -1,2 +1,133 @@
 pub const DISPLAY_WIDTH: usize = 240;
 pub const DISPLAY_HEIGHT: usize = 160;
+pub const PALETTE_SIZE: usize = 16;
+
+pub const BG_PALETTE_ADDRESS: u32 = 0x05000000;
+pub const OBJ_PALETTE_ADDRESS: u32 = 0x05000200;
+
+use std::{cell::RefCell, rc::Rc};
+
+#[cfg(feature = "debug")]
+use rand::Rng;
+
+use crate::{
+    internal_memory::InternalMemory,
+    render::palette_color::{PaletteColor, PaletteType},
+};
+
+pub struct PixelProcessUnit {
+    internal_memory: Rc<RefCell<InternalMemory>>,
+}
+
+impl PixelProcessUnit {
+    pub fn new(internal_memory: Rc<RefCell<InternalMemory>>) -> Self {
+        Self { internal_memory }
```

Comment:
In other iteration I would like to see this referring only to the part of video and not to entire `InternalMemory`, not important now, maybe we can just spit "logical memory" from "video memory"

Reply:
I agree :)